tests/test_introspection.py

```diff
@@ -145,3 +150,151 @@ def test_default_interfaces():
     # just make sure it doesn't throw
     default = intr.Node.default()
     assert type(default) is intr.Node
+
+
+class MockMessageBus(BaseMessageBus):
+    def __init__(self, nodes: dict[str, dict[str, intr.Node]]) -> None:
+        super().__init__(ProxyObject=MockProxyObject)
+        self.nodes = nodes
+        self.introspect_count = 0
+
+        # disable bus name tracking for testing purposes
+        class MockNameOwners(dict):
+            def get(self, key, default):
+                return dict.get(self, key, default if default else ":")
+
+        self._name_owners = MockNameOwners()
```

Comment:
I'm sorry, I just don't know how to appease Cython here. I'm subclassing `dict` so I can defeat the name tracking in `BaseProxyObject.get_interface(…)`, but Cython says it's expecting a `dict`, not a `MockNameOwners`. This is beyond my expertise.

Reply:
Cython is going to strongly enforce types. Can you mock it in a different way?